### PR TITLE
Add special DOM props to JSX

### PR DESCRIPTION
**Pull request by nickmessing**
Atomatically bind special props like `value` to `domProps` just like vue-template.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -273,6 +273,18 @@ describe('babel-plugin-transform-vue-jsx', () => {
...
   })
+
+  it('should handle special properties', () => {
+    const noop = _ => _
+    const vnode = render(h => (
+      <input value="value" />
+    ))
+    expect(vnode.data.domProps.value).to.equal('value')
+    const vnode2 = render(h => (
+      <input value="value" type="button" />
```
Comment (yyx990803) [inline test/test.js]: Just occurs to me that a static `value` attribute should probably be treated as the initial value only and thus be transpiled into `attrs`. Is there a way to differentiate this with `value={...}`? Probably just checking if the node's bound value is a literal string?

Comment (nickmessing) [inline test/test.js]: Yep, will do that in a second

**Comment by nickmessing** [comment]
@yyx990803, updated, should I add same behaviour for `babel-helper-vue-jsx-merge-props`?

**Comment by yyx990803** [comment]
@nickmessing yes

**Comment by nickmessing** [comment]
@yyx990803, I don't see how we can use it there since we don't even support `<a {...{onClick: this.fn}} />`, it's just merging objects. Maybe we should consider runtime attrs parser similar to one included in here?

**Comment by yyx990803** [comment]
Released in 3.5.0

**Pull request merged**